elixir/lib/card_deck.ex

```diff
@@ -0,0 +1,31 @@
+defmodule PlayingCards.CardDeck do
+  alias PlayingCards.Card
+
+  @cards (
+           cards =
+             for suit <- ~w(♠︎ ♣︎ ♥︎ ♦︎),
+                 rank <- ~w(A 2 3 4 5 6 7 8 9 10 J Q K),
+                 do: %Card{suit: suit, rank: rank}
+
+           jokers = for joker <- ~w(Joker Joker), do: %Card{suit: "*", rank: joker}
+           cards ++ jokers
```

Comment:
you usually want to avoid concatenating lists if you can avoid it and prepend wherever possible - as it's a potentially expensive operation with linked lists that Erlang/Elixir uses for everything.

This is (very slightly) better performing, but much less readable, and seeing as this is done at compile time, pretty negligible in the scheme of things.

```suggestion
           joker = %Card{suit:"*", rank:"Joker"}
           [joker | [joker | cards]]
```

Reply:
Good to know though, I had heard about the performance aspect of working with lists in Elixir but had forgotten the advice about it 😂 as you say looks a little more cryptic in this instance. Thanks for the pointer.